src/components/map/mapbox/controls/DateRangeControl/DateRangeControl.tsx

```diff
@@ -0,0 +1,95 @@
+import mapboxgl, { IControl, Map } from "mapbox-gl";
+import { createRoot, Root } from "react-dom/client";
+import React, { useContext, useEffect, useMemo, useState } from "react";
+import MapContext from "../../MapContext";
+import DateSliderControlClass from "./DateSliderControl";
+import timeRange from "../../../../../../src/assets/images/time-range.png";
+import {
+  DownloadConditionType,
+  IDownloadCondition,
+} from "../../../../../pages/detail-page/context/DownloadDefinitions";
+
+class DateRangeControlClass implements IControl {
+  private container: HTMLDivElement | null = null;
+  private iconRoot: Root | null = null;
+  private map: Map | undefined | null = undefined;
+
+  private readonly setIsShowingDateSelector: () => void;
+
+  constructor(setIsShowingDateSelector: () => void) {
+    this.setIsShowingDateSelector = setIsShowingDateSelector;
+  }
+
+  onAdd(map: mapboxgl.Map): HTMLElement {
+    this.map = map;
+    this.container = document.createElement("div");
+    this.container.className = "mapboxgl-ctrl-group mapboxgl-ctrl";
+    this.iconRoot = createRoot(this.container);
+    this.iconRoot.render(
+      <button>
+        <img alt="" src={timeRange}></img>
+      </button>
+    );
+    this.container.onclick = this.setIsShowingDateSelector;
+    return this.container;
+  }
+
+  onRemove(_: mapboxgl.Map) {
```

Comment:
Please refer to ToggleControl.tsx and how to unmount and discard control properly, if you can refactor to make it common so that all control call the same discard will be even better

Reply:
👌  thank you. On it